elmslie-android/src/main/java/vivid/money/elmslie/android/base/ElmFragment.kt

```diff
@@ -18,6 +19,23 @@ abstract class ElmFragment<Event : Any, Effect : Any, State : Any> : Fragment,
     @Suppress("LeakingThis", "UnusedPrivateMember")
     private val elm = ElmScreen(this, lifecycle) { requireActivity() }
 
+    override val isAllowedToRunMvi: Boolean
+        get() {
+            var currentFragment: Fragment? = this
```

Comment:
I don't really know how to put it for you, but I still don't really get the use case for checking the parent hierarchy. Here's how i understand it. Right now all of the fragment in the authorized zone must update this property to false. Most likely to handle this automatically it will be set to false in the BaseFragment and all of the fragments in the not authorized zone will override it to true. But in this case it will be false in all of the child fragments as well (as they inherit from BaseFragment). So this while loop will never even be launched.

Reply:
I set this property to false only in MainFragment and call navigation event right after it, it will give me a full control over starting Elm. I set this property only when I really need to, when I'm sure that parent and all child fragments will be closed.

Reply:
Here is an example

```
class MainFragment : BaseFragment<MainScreenEvent, MainScreenEffect, MainScreenState>() { 

    override fun onCreate(savedInstanceState: Bundle?) {
            if (ProcessDeathDetector.isRestoringAfterProcessDeath) {
                // here we detect that navigation event will be take place and set isAllowedToRunMvi = false
                onRestoringAfterProcessDeath()
            }

            // here all child onCreate happend and we already have isAllowedToRunMvi = false
            // and all child stores will not be started at all
            super.onCreate(savedInstanceState) 

           .
    }

    private fun onRestoringAfterProcessDeath() {
        navigator.openNewRootScreen(AuthFragment::class.java)
        isAllowedToRunMvi = false
    }

}
```

Reply:
So in fact, we set isAllowedToRunMvi when we are sure that parent and children fragments will be recycled right after restoring.